packages/esm-patient-search-app/src/hooks/useArrowNavigation.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState, useCallback } from 'react';
+
+const useArrowNavigation = (
+  inputRef: React.MutableRefObject<HTMLInputElement>,
+  totalResults: number,
+  enterCallback: (evt: any, index: number) => void,
+  resetFocusCallback: () => void,
+  initalFocussedResult: number = -1,
+) => {
+  const [focussedResult, setFocussedResult] = useState(initalFocussedResult);
+
+  const handleKeyPress = useCallback(
+    (e) => {
+      if (e.key === 'ArrowUp') {
+        setFocussedResult((prev) => Math.max(-1, prev - 1));
+      } else if (e.key === 'ArrowDown') {
+        setFocussedResult((prev) => Math.min(totalResults - 1, prev + 1));
+      } else if (e.key === 'Enter' && focussedResult > -1) {
+        enterCallback(e, focussedResult);
+      } else if (document.activeElement !== inputRef.current) {
+        resetFocusCallback();
+        setFocussedResult(initalFocussedResult);
+      }
+    },
+    [
+      setFocussedResult,
```

Comment:
Actually, when we do the linting, it gives the warning, hence I keep it for clear linting logs.
Thanks!

Reply:
I know. The linter only checks what is used and not in the array. It is, unfortunately, not clever enough to identify intention here (could be that something is dropped because you only want to capture the value at function creation) or optimizations such as this one.